test/utils.py

```diff
@@ -132,20 +132,44 @@ def capture(command, *args, stdout=subprocess.PIPE, encoding="utf-8", **kwargs):
     return run(command, *args, stdout=stdout, encoding=encoding, **kwargs).stdout
 
 
-def wait_until(error_message="Did not complete", timeout=5, interval=0.1):
+def wait_until(
+    error_message="Did not complete", timeout=5, interval=0.1, min_attempt_count=5
+):
     """
     Loop until the timeout is reached. If the timeout is reached, raise an
     exception with the given error message.
     """
-    start = time.time()
-    end = start + timeout
-    last_printed_progress = start
-    while time.time() < end:
-        if timeout > 5 and time.time() - last_printed_progress > 5:
-            last_printed_progress = time.time()
-            print(f"{error_message} in {time.time() - start} seconds - will retry")
+    start_ts = time.monotonic()
+    end_ts = start_ts + timeout
+    last_printed_progress = start_ts
+    attempt = 0
```

Comment:
If we put the yield here and remove `min_attempt_count` there's no reason to keep track of the number of attempts.
```suggestion
    yield
```

Reply:
> If we put the yield here and remove `min_attempt_count` there's no reason to keep track of the number of attempts.

1) You are confusing tests and the program being tested. I agree that in programs we should "optimize" the code but the test must help to understand what happens when they fail.

If this function will generate timeout exception we will see (in back-trace) how many attempts were made.

2) min_attempt_count was added to debug purposes. It may help on the slow machines, too.

Reply:
I kept the `attempt` counter, and included in the print.

> min_attempt_count was added to debug purposes. 

I don't understand how it helps with debugging tests. So I removed it for now. If you explain how it helps with debugging, maybe we can add it back.

Reply:
> > min_attempt_count was added to debug purposes.
> 
> I don't understand how it helps with debugging tests. So I removed it for now. If you explain how it helps with debugging, maybe we can add it back.

We test your pgbouncer on the really slow machines (of course, on the fast machines, too).

Default pytest-timeout was increased there but it does not help.

min_attempt_count give an additional chance to finish this unstable test on the such conditions.